### ISSUE TITLE
spiceinit failed with Chandrayann-1 M3 scene

### ISSUE THREAD

**Issue by whfarrand05** [DOI-USGS/ISIS3#5420]
**ISIS version(s) affected**: 4.2.0

**Description**  
On my Mac, I tried to run spiceinit on a M3 scene with the following command:
spiceinit from=M3G20090112T165924_V03.cub 

and I got the following error message:
**ERROR** No existing files found with a numerial version matching [kernels.????.db] in [/opt/anaconda3/envs/isis/data/chandrayaan1/kernels/pck].

I am not really sure if I was able to successfully download all the M3 spice files since that process didn't seem to take very long, and I recall from downloading LRO kernels that it took a long time so perhaps there was an issue with the download???

**Comment by AustinSanders** [comment]
Hi @whfarrand05 ,

It is worth checking that the expected files are in your data area.  Can you run `ls /opt/anaconda3/envs/isis/data/chandrayaan1/kernels/pck` and verify that kernels.0001.db is present?

**Comment by whfarrand05** [comment]
No, that file isn't there.  This is all that's in that ~/pck subdirectory:


-rw-r--r--  1 wfarrand  wheel   111586 Feb 10  2009 PCK00008.TPC

-rw-r--r--  1 wfarrand  wheel   126143 Oct 25  2011 PCK00010.TPC

-rw-r--r--  1 wfarrand  wheel  1770496 Jan 14  2009 moon_pa_de421_1900-2050.bpc

I had a lot of trouble getting the DownloadIsisData script to run... Maybe you can let me know what you think the best way to run it is on a Mac with Anaconda.

Thanks,
Bill

 ---------------------------------

 Wm. Farrand, PhD
 Senior Research Scientist
 Space Science Institute
 4765 Walnut Street, Suite B
 Boulder CO 80301
http://www.spacescience.org<https://mail.spacescience.org/owa/redir.aspx?C=b80995152d0645bd9285442005390dae&URL=http%3a%2f%2fwww.spacescience.org>


“Every great advance in natural knowledge has involved the absolute rejection of authority.” -- T. H. Huxley, 1870
________________________________

**Comment by Kelvinrr** [comment]
If the PCK directory is empty it might cause issues with the kernel search algorithm. IIRC this was fixed in a more recent version of ISIS. If you don't want to upgrade, simply deleting the empty PCK folder should fix it.

**Comment by whfarrand05** [comment]
Thanks. I decided to install ISIS on another Mac to make sure I got the latest version as suggested. I’m having issues again with the downloadIsisData script. Pasted below is the command issued and the response. So what is best way to get that script to run?



./downloadIsisData chandrayaan1 $ISISDATA

zsh: permission denied: ./downloadIsisData



From: Kelvin Rodriguez ***@***.***>
Date: Thursday, February 8, 2024 at 3:35 PM
To: DOI-USGS/ISIS3 ***@***.***>
Cc: Bill Farrand ***@***.***>, Mention ***@***.***>
Subject: Re: [DOI-USGS/ISIS3] spiceinit failed with Chandrayann-1 M3 scene (Issue #5420)
You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

If the PCK directory is empty it might cause issues with the kernel search algorithm. IIRC this was fixed in a more recent version of ISIS. If you don't want to upgrade, simply deleting the empty PCK folder should fix it.

**Comment by lwellerastro** [comment]
FYI - the local  and efs isis8.1.0-RC2 data area does have a populated pck directory
ls -1 $ISISDATA/chandrayaan1/kernels/pck/
kernels.0001.db
moon_080317.tf
moon_assoc_me.tf
moon_pa_de421_1900-2050.bpc
PCK00008.TPC
PCK00010.TPC

These are the kernels I pick up for this image:
 ```
 Group = Kernels
    NaifFrameCode             = -86520
    LeapSecond                = $base/kernels/lsk/naif0012.tls
    TargetAttitudeShape       = ($base/kernels/pck/pck00009.tpc,
                                 $chandrayaan1/kernels/pck/moon_080317.tf,
                                 $chandrayaan1/kernels/pck/moon_assoc_me.tf)
    TargetPosition            = ($Chandrayaan1/kernels/tspk/moon_pa_de421_190-
                                 0-2050.bpc,
                                 $Chandrayaan1/kernels/tspk/de421.bsp)
    InstrumentPointing        = $chandrayaan1/kernels/ck/M3G20090112T165924_V-
                                03_L1B_nadir-jig_2016-04-29.bc
    Instrument                = Null
    SpacecraftClock           = $chandrayaan1/kernels/sclk/aig_ch1_sclk_compl-
                                ete_biased_m1p816.tsc
    InstrumentPosition        = $chandrayaan1/kernels/spk/M3G20090112T165924_-
                                V03_L1B_nadir-jig_2016-04-29.bsp
    InstrumentAddendum        = $chandrayaan1/kernels/iak/m3Addendum005.ti
    ShapeModel                = $base/dems/ldem_128ppd_Mar2011_clon180_radius-
                                _pad.cub
    InstrumentPositionQuality = Reconstructed
    InstrumentPointingQuality = Reconstructed
    CameraVersion             = 1
    Source                    = ale
  End_Group
```

**Comment by whfarrand05** [comment]
When I did the same:
ls -1 $ISISDATA/chandrayaan1/kernels/pck/

I just got the following:
moon_pa_de421_1900-2050.bpc
PCK00008.TPC
PCK00010.TPC

Should I just try a re-install of the Chandrayaan-1 data, or could the missing files have gone elsewhere?

Thanks,
Bill

From: lwellerastro ***@***.***>
Date: Friday, February 9, 2024 at 10:42 AM
To: DOI-USGS/ISIS3 ***@***.***>
Cc: Bill Farrand ***@***.***>, Mention ***@***.***>
Subject: Re: [DOI-USGS/ISIS3] spiceinit failed with Chandrayann-1 M3 scene (Issue #5420)
You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

FYI - the local and efs isis8.1.0-RC2 data area does have a populated pck directory
ls -1 $ISISDATA/chandrayaan1/kernels/pck/
kernels.0001.db
moon_080317.tf
moon_assoc_me.tf
moon_pa_de421_1900-2050.bpc
PCK00008.TPC
PCK00010.TPC

These are the kernels I pick up for this image:

Group = Kernels

   NaifFrameCode             = -86520

   LeapSecond                = $base/kernels/lsk/naif0012.tls

   TargetAttitudeShape       = ($base/kernels/pck/pck00009.tpc,

                                $chandrayaan1/kernels/pck/moon_080317.tf,

                                $chandrayaan1/kernels/pck/moon_assoc_me.tf)

   TargetPosition            = ($Chandrayaan1/kernels/tspk/moon_pa_de421_190-

                                0-2050.bpc,

                                $Chandrayaan1/kernels/tspk/de421.bsp)

   InstrumentPointing        = $chandrayaan1/kernels/ck/M3G20090112T165924_V-

                               03_L1B_nadir-jig_2016-04-29.bc

   Instrument                = Null

   SpacecraftClock           = $chandrayaan1/kernels/sclk/aig_ch1_sclk_compl-

                               ete_biased_m1p816.tsc

   InstrumentPosition        = $chandrayaan1/kernels/spk/M3G20090112T165924_-

                               V03_L1B_nadir-jig_2016-04-29.bsp

   InstrumentAddendum        = $chandrayaan1/kernels/iak/m3Addendum005.ti

   ShapeModel                = $base/dems/ldem_128ppd_Mar2011_clon180_radius-

                               _pad.cub

   InstrumentPositionQuality = Reconstructed

   InstrumentPointingQuality = Reconstructed

   CameraVersion             = 1

   Source                    = ale

 End_Group

**Comment by AustinSanders** [comment]
@whfarrand05 I do have a workaround that should allow you to get the kernels, but I'm not sure if it will have any side effects.

Locate your rclone.conf, which will be either in your home directory or your conda environment (at $CONDA_PREFIX/etc/isis/rclone.conf ) and change:

```
[chandrayaan1_usgs]
type = alias
remote = asc_s3:asc-isisdata/usgs_data/chandrayaan1/
```
to 
```
[chandrayaan1_usgs]
type = alias
remote = asc_s3:asc-isisdata/isis_data/chandrayaan1/
```

and run `downloadIsisData chandrayaan1  $ISISDATA --include="{pck/**}"`

**Comment by whfarrand05** [comment]
Thanks. I tried that, but it looks like it had problems with the downloadIsisData script.  Here’s what I got:


(isis) ***@***.*** scripts % ./downloadIsisData chandrayaan1 $ISISDATA --include="{pck/**}"

2024/02/12 11:03:00 ERROR : Local file system at /Users/wfarrand/anaconda3/envs/isis/data/chandrayaan1/kernels: Ignoring --track-renames as the source and destination do not have a common hash

2024/02/12 11:03:00 ERROR : Local file system at /Users/wfarrand/anaconda3/envs/isis/data/chandrayaan1/kernels: Ignoring --track-renames as it doesn't work with copy or move, only sync

Transferred:                   0 B / 0 B, -, 0 B/s, ETA -

Checks:                 3 / 3, 100%

Elapsed time:         7.7s

2024/02/12 11:03:08 ERROR : Local file system at /Users/wfarrand/anaconda3/envs/isis/data/chandrayaan1: Ignoring --track-renames as it doesn't work with copy or move, only sync

Transferred:           20.946 KiB / 20.946 KiB, 100%, 0 B/s, ETA -

Checks:                 3 / 3, 100%

Transferred:            2 / 2, 100%

Elapsed time:         1.0s


From: Austin Sanders ***@***.***>
Date: Monday, February 12, 2024 at 10:45 AM
To: DOI-USGS/ISIS3 ***@***.***>
Cc: Bill Farrand ***@***.***>, Mention ***@***.***>
Subject: Re: [DOI-USGS/ISIS3] spiceinit failed with Chandrayann-1 M3 scene (Issue #5420)
You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

@whfarrand05<https://github.com/whfarrand05> I do have a workaround that should allow you to get the kernels, but I'm not sure if it will have any side effects.

Locate your rclone.conf, which will be either in your home directory or your conda environment (at $CONDA_PREFIX/etc/isis/rclone.conf ) and change:

[chandrayaan1_usgs]

type = alias

remote = asc_s3:asc-isisdata/usgs_data/chandrayaan1/

to

[chandrayaan1_usgs]

type = alias

remote = asc_s3:asc-isisdata/isis_data/chandrayaan1/

and run downloadIsisData chandrayaan1 $ISISDATA --include="{pck/**}"

**Comment by AustinSanders** [comment]
Just to confirm, this is after editing your rclone.conf file?

The error you're getting should be specific to the --track-renames directive, but it shouldn't stop you from downloading the files.  I got the error when running the workaround, but the files were downloaded correctly.

**Comment by whfarrand05** [comment]
Yes. That is after making the change. I am wondering though, I found that rclone.conf file in the following directory in my system:


/Users/wfarrand/anaconda3/pkgs/isis-7.2.0-np124_0/etc/isis

However, most of the other files are in:


/Users/wfarrand/anaconda3/envs/isis

I wanted to make sure something nutty didn’t happen with the installation.

Thanks


From: Austin Sanders ***@***.***>
Date: Monday, February 12, 2024 at 11:12 AM
To: DOI-USGS/ISIS3 ***@***.***>
Cc: Bill Farrand ***@***.***>, Mention ***@***.***>
Subject: Re: [DOI-USGS/ISIS3] spiceinit failed with Chandrayann-1 M3 scene (Issue #5420)
You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

Just to confirm, this is after editing your rclone.conf file?

**Comment by whfarrand05** [comment]
Any more suggestions on what might be the issue?

I had edited the rclone.conf file as suggested but still seem to be missing files in the sub-directories of the ~/chandrayaan1/kernels directory.

Bill


From: Austin Sanders ***@***.***>
Date: Monday, February 12, 2024 at 11:12 AM
To: DOI-USGS/ISIS3 ***@***.***>
Cc: Bill Farrand ***@***.***>, Mention ***@***.***>
Subject: Re: [DOI-USGS/ISIS3] spiceinit failed with Chandrayann-1 M3 scene (Issue #5420)
You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

Just to confirm, this is after editing your rclone.conf file?

**Comment by AustinSanders** [comment]
I cannot reproduce the issue when targeting isis_data in the rclone configuration file. 

My only suggestion is running `rclone config file` to ensure that you're modifying the config that you're using.

**Comment by whfarrand05** [comment]
I guess I’m unclear of whether using the rclone approach is a necessary step in a path or just an alternative to running the downloadIsisData script? I was able to do the latter successfully. However, then after doing that and trying “spiceinit” for the file I’m working with, I get:


spiceinit from=M3G20090112T165924_V03.cub

**ERROR** No existing files found with a numerical version matching [kernels.????.db] in [/Users/wfarrand/anaconda3/envs/isis/data/chandrayaan1/kernels/iak].

I go into the ~ data/chandrayaan1/kernels directory and don’t see any subdirectory “iak”. So it seems like some stuff was downloaded and some stuff wasn’t?

Thanks,
Bill



From: Austin Sanders ***@***.***>
Date: Thursday, February 15, 2024 at 7:48 PM
To: DOI-USGS/ISIS3 ***@***.***>
Cc: Bill Farrand ***@***.***>, Mention ***@***.***>
Subject: Re: [DOI-USGS/ISIS3] spiceinit failed with Chandrayann-1 M3 scene (Issue #5420)
You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

I cannot reproduce the issue when targeting isis_data in the rclone configuration file.

My only suggestion is running rclone config file to ensure that you're modifying the config that you're using.

**Comment by AustinSanders** [comment]
I'm not sure which step you're referring to as "the rclone approach."  If you're just referring to editing the rclone configuration file, then no, it is not an alternative to downloadIsisData, it's just changing the remote directory that is searched by downloadIsisData.

When I suggested running `downloadIsisData chandrayaan1  $ISISDATA --include="{pck/**}"`, I was under the impression that you were only missing pck files.  What happens if, after editing your config, you run `downloadIsisData chandrayaan1  $ISISDATA ` ?
